src/components/Header.svelte

```diff
@@ -66,22 +77,32 @@
           <img class="logo" src="/icon.png" alt="Fission username" />
           <span>{session.username}</span>
         </div>
-        <div class="filecoin-indicator">
-          <img
-            class="logo"
-            src="/filecoin-logo.svg"
-            alt="Filecoin wallet balance"
-          />
-          <span>10.0 FIL</span>
-        </div>
-        <div class="filecoin-indicator">
-          <img
-            class="logo"
-            src="/filecoin-symbol-color.svg"
-            alt="Lotus provider balance"
-          />
-          <span>250.3 FIL</span>
-        </div>
+        {#if wallet !== undefined}
+          <div class="filecoin-indicator">
+            {#await wallet?.getBalance() then balance}
```

Comment:
Is this typical of svelte to make blocking calls like this in the markup? seems weird... also does this just not render until getBalance completes?

Reply:
Yep, rendering happens when `getBalance` completes. There is an alternative syntax for showing a loading message instead of not rendering anything. Here is an example that shows that: https://svelte.dev/tutorial/await-blocks.

The await blocks are a bit contentious. Some folks like them, some don't. My strategy has been to use them when it is a one-off thing in a component, like the result is only displayed in one place.